src/containers/Tag/TagList/TagList.tsx

```diff
@@ -53,67 +121,154 @@ export const TagList: React.SFC<TagListProps> = (props) => {
     return <Redirect to="/tag/add" />;
   }
 
-  if (loading) return <p>Loading...</p>;
-  if (error) return <p>Error :(</p>;
-  const tagList = data.tags;
+  if (loading || l) return <p>Loading...</p>;
+  if (error || e) return <p>Error :(</p>;
 
   const deleteHandler = (id: number) => {
     deleteId = id;
     deleteTag({ variables: { id } });
     setNotification(client, 'Tag deleted Successfully');
   };
 
-  let listing: any;
-  if (tagList.length > 0) {
-    listing = tagList.map((n: any) => {
+  // Reformat all tags to be entered in table
+  function getIcons(id: number | undefined) {
+    if (id) {
       return (
-        <TableRow key={n.id}>
-          <TableCell component="th" scope="row">
-            {n.label}
-          </TableCell>
-          <TableCell scope="row">{n.description}</TableCell>
-          <TableCell>
-            <Link to={'/tag/' + n.id + '/edit'}>
-              <IconButton aria-label="Edit" color="default">
-                <EditIcon />
-              </IconButton>
-            </Link>
-            <IconButton aria-label="Delete" color="default" onClick={() => deleteHandler(n.id!)}>
-              <DeleteIcon />
+        <>
+          <Link to={'/tag/' + id + '/edit'}>
+            <IconButton aria-label="Edit" color="default">
+              <EditIcon />
             </IconButton>
-          </TableCell>
-        </TableRow>
+          </Link>
+          <IconButton aria-label="Delete" color="default" onClick={() => deleteHandler(id!)}>
+            <DeleteIcon />
+          </IconButton>
+        </>
       );
+    }
+  }
+
+  function formatTags(tags: Array<any>) {
+    // Should be type tag, but can't import Tag type into file
+    return tags.map((t: any) => {
+      return {
+        label: t.label,
+        description: t.description,
+        operations: getIcons(t.id),
+      };
     });
-  } else {
-    listing = (
-      <TableRow>
-        <TableCell>There are no tags.</TableCell>
-      </TableRow>
-    );
+  }
+
+  const resetTableVals = () => {
+    setTableVals({
+      pageNum: 0,
+      pageRows: 10,
+      sortCol: columnNames[0],
+      sortDirection: 'asc',
+    });
+  };
+
+  const handleSearch = (e: any) => {
+    e.preventDefault();
+    let searchVal = e.target.nameSearch.value.trim();
+    setSearchVal(searchVal);
+    resetTableVals();
+  };
+
+  // Making different columns for the table
+  const oldColumns = [
```

Comment:
Not needed with currently pagination implementation. Will be removed later (just used for testing purposes with client-side pagination component).